src/index.ts

```diff
@@ -0,0 +1,141 @@
+import { connect, WebSocketOptions } from 'it-ws/client'
+import { multiaddrToUri as toUri } from '@multiformats/multiaddr-to-uri'
+import { AbortError } from '@libp2p/interfaces/errors'
+import pDefer from 'p-defer'
+import debug from 'debug'
+import env from 'wherearewe'
+import { createListener } from './listener.js'
+import { socketToMaConn } from './socket-to-conn.js'
+import * as filters from './filters.js'
+import type { Transport, Upgrader, MultiaddrFilter } from '@libp2p/interfaces/transport'
+import type { AbortOptions } from '@libp2p/interfaces'
+import type { WebSocketListenerOptions } from './listener.js'
+import type { Multiaddr } from '@multiformats/multiaddr'
+import type { DuplexWebSocket } from 'it-ws/dist/src/duplex'
+
+const log = Object.assign(debug('libp2p:websockets'), {
+  error: debug('libp2p:websockets:error')
+})
+
+/**
+ * @class WebSockets
+ */
+export class WebSockets implements Transport<AbortOptions & WebSocketOptions, WebSocketListenerOptions> {
+  private readonly upgrader: Upgrader
+  private readonly _filter?: MultiaddrFilter
+
+  constructor (opts: { upgrader: Upgrader, filter?: MultiaddrFilter }) {
+    const { upgrader, filter } = opts
+
+    if (upgrader == null) {
+      throw new Error('An upgrader must be provided. See https://github.com/libp2p/interface-transport#upgrader.')
```

Comment:
This and another reference to the interface-transport repo, should reference js-libp2p-interfaces instead